public_html/script/src/graphical-util.js

```diff
@@ -0,0 +1,138 @@
+        if (args.length === 2) {
+            pt = new Point(args[0], args[1]);
+        }
+        return (pt.x >= this.x && pt.x <= this.right &&
+                pt.y >= this.y && pt.y <= this.bottom);
+    }
+};
+
+export function aspectFitRect(outsideRect, widthToHeightRatio) {
+    const returnFrame = new Rect();
+    if (outsideRect.width / outsideRect.height > widthToHeightRatio) {
+        //returned frame will have the same height as outsideRect
+        returnFrame.height = outsideRect.height;
+        returnFrame.width = returnFrame.height * widthToHeightRatio;
+        returnFrame.x = outsideRect.x + outsideRect.width / 2 - returnFrame.width / 2;
+        returnFrame.y = outsideRect.y;
+    } else {
+        //returned frame will have the same width as outsideRect
+        returnFrame.width = outsideRect.width;
+        returnFrame.height = returnFrame.width / widthToHeightRatio;
+        returnFrame.y = outsideRect.y + outsideRect.height / 2 - returnFrame.height / 2;
+        returnFrame.x = outsideRect.x;
+    }
+    return returnFrame;
+}
+
+export function aspectFillRect(outsideRect, widthToHeightRatio) {
+    const returnFrame = new Rect();
+    if (outsideRect.width / outsideRect.height > widthToHeightRatio) {
+        //returned frame will have the same width as outsideRect
+        returnFrame.width = outsideRect.width;
+        returnFrame.height = returnFrame.width / widthToHeightRatio;
+        returnFrame.y = outsideRect.y + outsideRect.height / 2 - returnFrame.height / 2;
+        returnFrame.x = outsideRect.x;
+    } else {
+        //returned frame will have the same height as outsideRect
+        returnFrame.height = outsideRect.height;
+        returnFrame.width = returnFrame.height * widthToHeightRatio;
+        returnFrame.x = outsideRect.x + outsideRect.width / 2 - returnFrame.width / 2;
+        returnFrame.y = outsideRect.y;
+    }
+    return returnFrame;
+}
+
+export function linesFromString({string, fontSize, frame}) { //truncates words that are too long
+    setFont({size: fontSize});
+    const lines = [];
+    let spaceIndex, currentLine = "", previousIndex = -1;
+    while ((spaceIndex = string.indexOf(" ", previousIndex + 1)) !== -1) {
+        const word = string.substring(previousIndex + 1, spaceIndex);
+        const [clWidth, wordWidth] = [textWidth(currentLine), textWidth(word)];
+        if (clWidth + wordWidth > frame.width) {
+            lines.push(currentLine.trim());
+            currentLine = word;
+            if (wordWidth > frame.width) {
+                lines.push(currentLine.substr(0, Math.round(frame.width / wordWidth)));
+                currentLine = "";
+            }
+        }
+        else {
+            currentLine += ' ' + word;
+        }
+        previousIndex = spaceIndex;
+    }
+    currentLine += ' ' + string.substr(string.lastIndexOf(' ') + 1);
+    lines.push(currentLine);
+    return lines;
+}
+
+export function dist(a, b) { //a, b are points
```

Comment:
Maybe add this as a method to `Point`? `a.dist(b)` or something? Can keep this one also